crawl-ref/source/item-use.cc

```diff
@@ -3012,6 +3012,14 @@ bool drink(item_def* potion)
     // Check for Delatra's gloves before potentially melding them.
     bool heal_on_id = player_equip_unrand(UNRAND_DELATRAS_GLOVES);
 
+    // Drinking with hostile visible mons nearby resets unrand "Victory" stats.
+    // We need to check for this before potentially melding it.
+    if (player_equip_unrand(UNRAND_VICTORY)
+        && there_are_monsters_nearby(true, true, false))
+    {
+        you.props[VICTORY_CONDUCT_KEY] = true;
+    }
+
```

Comment:
Hmm, looking into this a bit further it also looks like this introduces some new inconsistencies - if you abort quaffing a potion for some reason (e.g. after being prompted to go berserk while wielding a non-weapon, or to go invisible while backlit), it'll now reset the unrand stats due to this being moved to before `quaff_potion()`.

Reply:
Thanks for the feedback kate, I'll take another look at this.